### ISSUE TITLE
Output `mop setup` yaml files cannot be used by `mop setup`

### ISSUE THREAD

**Issue by dsroberts** [ACCESS-Community-Hub/ACCESS-MOPPeR#112]
When running `mop setup`, mopper helpfully outputs a yaml file of the configuration used. This yaml file cannot be passed back into `mop setup` as it writes absolute paths to `tables_path` and `master_map`, where the application accepts only relative paths. Its probably a good idea to use [pathlib](https://docs.python.org/3/library/pathlib.html) instead of string concatenation operations as this handles operations on both relative and absolute paths without issue. Then it'd be fine to write absolute paths only in the output yaml file.

**Comment by paolap** [comment]
Thanks, I'll think about it as there are some differences between the two files and as everything get read form the config files, a user would still have to know what to remove potentially from the final output config. Not sure about this I need to check if actually would cause issues.

**Comment by paolap** [comment]
I mean I can change to path lib, just not sure if the output config can be used as it is to rerun the experiment. It was meant to be there for provenance never tried to rerun exactly the same thing.

**Comment by dsroberts** [comment]
The only reason I found this out is that when switching between the doi version and the main branch, I forgot the name of the config file I was using, and just picked the most sensibly named one. This turned out to be the one written by `mopdb template`, that does work with `mop run`. I wasn't aware this was never intended to work.

**Comment by paolap** [comment]
It could be useful if you want to rerun exactly an experiment it was just something I haven't yet fully considered as it was low priority. I'm moving to path lib regardless and then we can test that

**Comment by paolap** [comment]
I'm now using pathlib for mop setup. This works however the yaml config file is getting paths instead of strings and writing them as:

  outpath: !!python/object/apply:pathlib.PosixPath
  - /
  - scratch
  - v45
  - pxp581
  - MOPPER_output
  - sebcm2test

So I still have to fix that and convert all the paths to string when writing them

**Comment by paolap** [comment]
Finally the output config files can be used to reproduced the same config env.
I guess this helps if the first version is lost and the "provenance" one has been kept.

Still I'm not sure I used pathlib as you intended for the following reasons:
1) I can write pathlib to the yaml file but it causes and error when reading them (I might have to do something clever there ??
2) the way the code was written it doesn't matter how the derived paths are written in the file, as the code is ignoring them anyway, the only ones that get used are "appdir" and "outpath" which were absolute paths in both versions.
What I had to do was to modify a bit how "outpath" is generated so I'm adding the "exp_id" only if it's not already present in the path string. This produces correct paths.
